### PR TITLE
Fixing race bug

### PR DESCRIPTION
**Pull request by aanm**

### CODE REVIEW

File: daemon/daemon/endpoint.go [hunk 1]
```diff
@@ -588,8 +583,6 @@ func (d *Daemon) EndpointLabelsUpdate(epID uint16, op types.LabelOP, labels type
 				return fmt.Errorf("label %s not found, please add it first in order to enable it", v)
 			}
 		}
-		d.containersMU.Unlock()
```
Comment (tgraf) [inline daemon/daemon/endpoint.go:591]: This looks wrong. Why is this Unlock dropped?

Comment (aanm) [inline daemon/daemon/endpoint.go:591]: Was moved to line 594, 597, 602, 605.

Comment (tgraf) [inline daemon/daemon/endpoint.go:591]: Oh, it's in the other commit. This is what confused me.

**Comment by tgraf** [comment]
Single hunk in wrong commit, otherwise LGTM

File: daemon/daemon/endpoint.go (hunk 1 above)
Comment (aanm) [inline daemon/daemon/endpoint.go:591]: Oh sorry, somehow that slip through me, I'll move that to the proper commit

**Pull request merged**